moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -796,12 +795,10 @@ bool ServoCalcs::enforceSRDFPositionLimits()
 // Is handled differently for position vs. velocity control.
 void ServoCalcs::suddenHalt(trajectory_msgs::JointTrajectory& joint_trajectory)
 {
-  if (joint_trajectory.points.empty())
-  {
-    joint_trajectory.points.push_back(trajectory_msgs::JointTrajectoryPoint());
-    joint_trajectory.points[0].positions.resize(num_joints_);
-    joint_trajectory.points[0].velocities.resize(num_joints_);
-  }
+  joint_trajectory.points.clear();
+  joint_trajectory.points.push_back(trajectory_msgs::JointTrajectoryPoint());
+  joint_trajectory.points[0].positions.resize(num_joints_);
+  joint_trajectory.points[0].velocities.resize(num_joints_);
```

Comment:
I see. The failure mode would be a `joint_trajectory` with a point that had only the positions allocated, but not the velocities (or vice versa). Makes sense to flush the trajectory every time this is called.